README.md

```diff
@@ -111,6 +124,268 @@ instructions in its
 (coordinating guidance is in comments in the
 default recipe).
 
+
+## Other Guidelines
+
+### Modules and classes in cookbooks
+
+It is often useful to factor out logic into a library - especially logic that
+doesn't create resources. Doing so makes this logic easier to unit test and
+makes the recipe or resource cleaner.
+
+Our standard is that all cookbooks use the top-level container of `module FB`,
+and then create a class for their cookbook under that. For example, `fb_fstab`
+creates a `class Fstab` inside of the `module FB`. We will refer to this as
+the cookbook class from here.
+
+We require all cookbooks use this model for consitency.
+
+Since we don't put anything other than other classes inside the top-level
+object, it's clear that a `module` is the right choice.
+
+While there is no reason that a cookbook class can't be one design to be
+instantiated, more often than not it is simply a collection of callable class
+methods - a place to put methods that can then be called both from this
+cookbook and others.
+
+Below the cookbook class, the author is free to make whatever class or methods
+they desire.
+
+It is often desirable to `include` a module into a Custom Resource, and we do
```

Comment:
I don't know if I'd say this is "often" desirable -- while it's definitely ok if it's done for a reason, I find in many cases it makes things needlessly more complex.

Reply:
Are you responding to moving the logic into a module  or to `include`ing the module?

WRT to the first, I **strongly** disagree. For anything short of the simplest providers, it's the only way to make it reasonably unit-tested. Otherwise you end up mocking a ton of stuff in chefspec... and while I likes chefspec, it's far far far far easier to rspec-test a class than it is to chefspec test a provider, *especially* a provider in our model which is an "internal" provider.

Starting by sticking your methods in a class is far far simpler and will lead to cleaner code.

Now, wrt to the `include`ing, it gets a bit fuzzier. Bty default is' the same as python's multiinheritence, but that also is a bit implicit, so what I've been doing recently is:

```
action_class do
  class FbThingHelper
    include FB::Thing::Provider
  end
end
```

So that at least then all the methods are `FbThingHelper.whatever()` so it's at least clear where they come from.

Reply:
I rewrote this...

Reply:
Yeah, my concern was specifically about the `include`'ing, which I've found confusing to explain to people in the past. I like the usage of a helper class like this though, I think it makes the separation clearer and at least avoid polluting the namespace.